router/core/transport.go

```diff
@@ -84,68 +96,159 @@ func (ct *CustomTransport) RoundTrip(req *http.Request) (*http.Response, error)
 	return resp, err
 }
 
+func isRunningInsideDocker() bool {
+	// Check if we are running inside docker by
+	// testing by checking if /.dockerenv exists
+	st, err := os.Stat("/.dockerenv")
+	return err == nil && !st.IsDir()
+}
+
+// localhostFallbackRoundTripper is an http.RoundTripper that will retry failed
+// requests to localhost by rewriting the request to use is targetHost. Only
+// requests that fail with ECONNREFUSED will be retried.
+type localhostFallbackRoundTripper struct {
```

Comment:
Please make name it like `DockerHostTransport` to make it clear it is related to docker only.

Reply:
It's not Docker specific, it takes a field to indicate the target host.